src/domain/cpu/workers/hello.worker.ts

```diff
@@ -0,0 +1,63 @@
+export interface DataInMessagePayload {
+  offset: number;
+  chunkLength: number;
+  pattern: Uint8Array;
+  text: Uint8Array;
+}
+
+export interface DataOutMessagePayload {
+  matches: Array<number>;
+}
+
+class SolverCPU {
+  private readonly ctx: Worker = (self as unknown) as Worker;
+
+  constructor() {
+    this.ctx.onmessage = this.handleMessage.bind(this);
+  }
+
+  handleMessage(message: MessageEvent<DataInMessagePayload>) {
+    if (message.data.pattern.length < 1 || message.data.text.length < 1) {
```

Comment:
We should validate offset and chunkLength also

Reply:
Added validation for offset and chunkLength.